imports/plugins/core/accounts/server/no-meteor/mutations/inviteShopMember.js

```diff
@@ -0,0 +1,140 @@
+
+const inputSchema = new SimpleSchema({
+  email: String,
+  groupId: String,
+  name: String,
+  shopId: String
+});
+
+/**
+ * @name accounts/inviteShopMember
+ * @memberof Mutations/Accounts
+ * @summary Invite new admin users (not consumers) to secure access in the dashboard to permissions
+ * as specified in packages/roles
+ * @param {Object} context - GraphQL execution context
+ * @param {Object} input - Necessary input for mutation. See SimpleSchema.
+ * @param {String} input.shopId - shop to invite user
+ * @param {String} input.groupId - groupId to invite user
+ * @param {String} input.email - email of invitee
+ * @param {String} input.name - name of invitee
+ * @return {Promise<Object>} with boolean of found new account === true || false
+ */
+export default async function inviteShopMember(context, input) {
+  inputSchema.validate(input);
+  const { collections, user: userFromContext, userHasPermission } = context;
+  const { Accounts, Groups, Shops, users } = collections;
+  const {
+    email,
+    groupId,
+    name,
+    shopId
+  } = input;
+
+  if (!userHasPermission(["reaction-accounts", "account/invite"], shopId)) {
+    throw new ReactionError("access-denied", "Access denied");
+  }
+
+  // we always use primary shop data, so retrieve this shop first with `Reaction` helper,
+  // and only query the `Shops` collection if shopId !== primaryShop._id
+  const primaryShop = await Shops.findOne({ shopType: "primary" });
+  if (!primaryShop) throw new ReactionError("not-found", "No primary shop found");
+
+  let shop = primaryShop;
+  if (primaryShop._id !== shopId) {
+    shop = await Shops.findOne({ _id: shopId });
+  }
+  if (!shop) throw new ReactionError("not-found", "No shop found");
+
+  const group = await Groups.findOne({ _id: groupId });
+  if (!group) throw new ReactionError("not-found", "No gr found");
+
+  // we don't allow direct invitation of "owners", throw an error if that is the group
+  if (group.name === "owner") {
+    throw new ReactionError("bad-request", "Cannot directly invite owner");
+  }
+
+  // check to see if invited user has an account
+  const invitedUser = await users.findOne({ "emails.address": email });
+
+  // if there is a userAccount, check to see if email has been verified
+  const matchingEmail = invitedUser && invitedUser.emails && invitedUser.emails.find((accountEmail) => accountEmail.address === email);
+  const isEmailVerified = matchingEmail && matchingEmail.verified;
+
+  // set variables to pass to email templates
+  const invitedByName = getCurrentUserName(userFromContext);
+  let dataForEmail;
+  let userId;
+  let templateName;
+
+  if (invitedUser) {
+    userId = invitedUser._id;
+  }
+
+  // if the user already has an account, send in informative email instead of an invite email
+  if (invitedUser && isEmailVerified) {
+    // make sure user has an `Account` - this should always be the case
+    const invitedAccount = await Accounts.findOne({ userId });
```

Comment:
Since you're just checking for existence and don't need any fields back, you could improve performance by adding `{ projection: { _id: 1 } }` options argument.